### PR TITLE
Create independent toggle styles

### PR DESCRIPTION
**Pull request by RVMendoza**
## What does this PR do?
Bringing in styles to create an independent toggle style

### Associated Issue
#2938 

### Code
 - [x] Build Component in HTML
 - [x] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: html/base/_links.scss [hunk 1]
```diff
@@ -41,7 +41,6 @@
   &:hover,
   &:active,
   &:focus {
-    transition: $sprk-link-transition;
```
Comment (guern1kn) [inline html/base/_links.scss]: Was this intentional?

File: html/components/toggle.stories.js [hunk 2]
```diff
@@ -23,18 +23,12 @@ export const defaultStory = () => {
...
-          sprk-b-TypeBodyThree
-          sprk-u-BareButton
-          sprk-u-TextCrop--none
-          sprk-b-Link
-          sprk-b-Link--has-icon
-          sprk-b-Link--simple
-        "
+        class="sprk-c-Toggle__trigger"
         data-sprk-toggle="trigger"
         href="#"
```
Comment (guern1kn) [inline html/components/toggle.stories.js]: Can remove this href.

File: html/base/_links.scss (hunk 1 above)
Comment (afebbraro) [reply html/base/_links.scss]: We don't wanna remove this one.

File: html/components/_toggle.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  // Font considerations
+  color: $sprk-color-body-three;
+  font-family: $sprk-font-family-body-three;
+  font-size: $sprk-font-size-body-three;
+  font-weight: $sprk-font-weight-body-three;
+  line-height: $sprk-line-height-body-three;
+
+  // Button User Agent Resets
+  background-color: transparent;
```
Comment (afebbraro) [inline html/components/_toggle.scss]: These need to be vars

File: html/components/_toggle.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+////
+
+// Future proofing classes that don't have styles right now but might later.
+// .sprk-c-Toggle
+// .sprk-c-Toggle__content
+
+
+.sprk-c-Toggle__trigger {
+  // TODO: abstract these variables
+  border-bottom: $sprk-link-has-icon-border-bottom;
```
Comment (afebbraro) [inline html/components/_toggle.scss]: These need to be toggle vars added to settings

File: html/components/_toggle.scss [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  border-bottom: $sprk-toggle-trigger-border-bottom;
+  text-decoration: $sprk-toggle-trigger-text-decoration;
+  transition: $sprk-toggle-trigger-transition;
+
+  // Font considerations
+  color: $sprk-toggle-trigger-color;
+  font-family: $sprk-toggle-trigger-font-family;
+  font-size: $sprk-toggle-trigger-font-size;
+  font-weight: $sprk-toggle-trigger-font-weight;
+  line-height: $sprk-line-height-body-three;
```
Comment (afebbraro) [inline html/components/_toggle.scss:24]: Missing toggle var here

File: html/components/_toggle.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  padding-bottom: $sprk-toggle-trigger-padding-bottom;
+  padding-left: $sprk-toggle-trigger-padding-left;
+
+  &:hover,
+  &:active,
+  &:focus {
+    border-bottom: $sprk-toggle-trigger-hover-border-bottom;
+    color: $sprk-toggle-trigger-hover-color;
+  }
+
```
Comment (afebbraro) [inline html/components/_toggle.scss:42]: Extra line

File: html/settings/_settings.scss [hunk 7]
```diff
@@ -1392,6 +1392,43 @@ $sprk-accordion-summary-active-text-decoration: none;
 // Toggle
 ///
 
+/// The top border style for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1395]:
```suggestion
/// The top border style for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 8]
```diff
@@ -1392,6 +1392,43 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-text-decoration: $sprk-link-text-decoration !default;
+/// The transition for Toggle Trigger
+$sprk-toggle-trigger-transition: $sprk-link-transition !default;
+/// The color for Toggle Trigger
+$sprk-toggle-trigger-color: $sprk-color-body-three !default;
+/// The font family for Toggle Trigger
+$sprk-toggle-trigger-font-family: $sprk-font-family-body-three !default;
+/// The font size for Toggle Trigger
+$sprk-toggle-trigger-font-size: $sprk-font-size-body-three !default;
+/// The font weight for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1413]:
```suggestion
/// The font weight for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 9]
```diff
@@ -1392,6 +1392,43 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-font-weight: $sprk-font-weight-body-three !default;
+/// The font weight for Toggle Trigger
+$sprk-toggle-trigger-line-height: $sprk-line-height-body-three !default;
+/// The active border bottom style for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1417]:
```suggestion
/// The active border bottom style for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 10]
```diff
@@ -1392,6 +1392,43 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-hover-border-bottom: $sprk-link-has-icon-hover-border-bottom !default;
+/// The active color for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1419]:
```suggestion
/// The active color for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 11]
```diff
@@ -1392,6 +1392,43 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-hover-color: $sprk-link-has-icon-hover-color-text !default;
+/// The background color for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1421]:
```suggestion
/// The background color for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 12]
```diff
@@ -1392,6 +1392,43 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-background-color: transparent !default;
+/// The padding top for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1423]:
```suggestion
/// The padding top for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 13]
```diff
@@ -1392,6 +1392,45 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-background-color: transparent !default;
+/// The padding top for Toggle Trigger.
+$sprk-toggle-trigger-padding-top: 0 !default;
+/// The padding right for Toggle Trigger.
+$sprk-toggle-trigger-padding-right: 0 !default;
+/// The padding bottom for Toggle Trigger.
+$sprk-toggle-trigger-padding-bottom: 0 !default;
+/// The padding left for Toggle Trigger.
+$sprk-toggle-trigger-padding-left: 0 !default;
+/// The padding left for Toggle Trigger
```
Comment (afebbraro) [inline html/settings/_settings.scss:1431]:
```suggestion
/// The text alignment for Toggle Trigger.
```

File: html/components/footer.stories.js [hunk 14]
```diff
@@ -264,7 +264,10 @@ export const defaultStory = () => {
 
             <div class="sprk-o-Stack__item">
               <div data-sprk-toggle="container">
-                <a class="sprk-b-TypeBodyFour sprk-b-Link sprk-b-Link--simple sprk-b-Link--has-icon sprk-u-FontWeight--normal" data-sprk-toggle="trigger" href="#nogo">
+                <a
+                  class="sprk-c-Toggle__trigger sprk-b-TypeBodyFour sprk-u-FontWeight--normal"
```
Comment (afebbraro) [inline html/components/footer.stories.js:268]: Are all these extra classes supposed to be on here?

Comment (afebbraro) [reply html/components/footer.stories.js:268]: sprk-b-TypeBodyFour sprk-u-FontWeight--normal  ?

File: html/package.json [hunk 15]
```diff
@@ -26,6 +26,7 @@
   "homepage": "https://github.com/sparkdesignsystem/spark-design-system",
   "dependencies": {
     "dom-slider": "2.1.0",
+    "focus-visible": "^5.0.2",
```
Comment (afebbraro) [inline html/package.json:29]: Lets hard set this at 5.0.2

File: html/settings/_settings.scss [hunk 16]
```diff
@@ -1392,6 +1392,49 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-font-family: $sprk-font-family-body-three !default;
+/// The font size for Toggle Trigger.
+$sprk-toggle-trigger-font-size: $sprk-font-size-body-three !default;
+/// The font weight for Toggle Trigger.
+$sprk-toggle-trigger-font-weight: $sprk-font-weight-body-three !default;
+/// The line height for Toggle Trigger.
+$sprk-toggle-trigger-line-height: $sprk-line-height-body-three !default;
+/// The active border bottom style for Toggle Trigger.
+$sprk-toggle-trigger-hover-border-bottom: $sprk-link-has-icon-hover-border-bottom !default;
+/// The active color for Toggle Trigger.
```
Comment (bobwise) [inline html/settings/_settings.scss:1419]:
```suggestion
/// The hover text color for Toggle Trigger.
```

File: html/settings/_settings.scss [hunk 17]
```diff
@@ -1392,6 +1392,49 @@ $sprk-accordion-summary-active-text-decoration: none;
...
+$sprk-toggle-trigger-color: $sprk-color-body-three !default;
+/// The font family for Toggle Trigger.
+$sprk-toggle-trigger-font-family: $sprk-font-family-body-three !default;
+/// The font size for Toggle Trigger.
+$sprk-toggle-trigger-font-size: $sprk-font-size-body-three !default;
+/// The font weight for Toggle Trigger.
+$sprk-toggle-trigger-font-weight: $sprk-font-weight-body-three !default;
+/// The line height for Toggle Trigger.
+$sprk-toggle-trigger-line-height: $sprk-line-height-body-three !default;
+/// The active border bottom style for Toggle Trigger.
```
Comment (bobwise) [inline html/settings/_settings.scss:1417]:
```suggestion
/// The hover border bottom style for Toggle Trigger.
```

**Pull request merged**